File: app/src/main/java/com/jemin/assignment/Network/ApiClient.java
package com.jemin.assignment.Network;


import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Created by JAMCOM on 2018-06-06.
 */

public class ApiClient {
    // 연결고리 BaseURL
    public static String BASE_URL ="https://openapi.naver.com";
    private static Retrofit retrofit = null;
    public static Retrofit getRetrofit(){
        if(retrofit == null){
            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }
}

File: app/src/main/java/com/jemin/assignment/Network/ApiClient.kt
package com.jemin.assignment.Network


import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object ApiClient {
    // 연결고리 BaseURL
    var BASE_URL = "https://openapi.naver.com"
    private var retrofit: Retrofit? = null
    fun getRetrofit(): Retrofit? {
        if (retrofit == null) {
            retrofit = Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()
        }
        return retrofit
    }
}